### ISSUE TITLE
Extract stage — stats and quality filters over an explicit file listing

### ISSUE THREAD

**Issue by CedrickArmel** [CedrickArmel/radiologist#183]
### Context

The first of the three stages. It replaces the "scan a directory, process everything" behavior
with "process exactly the files named in this listing", so a large corpus can be worked through
in batches and new arrivals can be processed without reprocessing what came before. It runs
today's Haralick GLCM plus lung-asymmetry extraction and today's IQR plus lung-out-of-frame
filters, and stops short of split assignment: its manifest carries stats and exclusion flags
with the split column left empty. Manifests from repeated extract runs accumulate side by side
in one destination folder; no manifest is ever appended to or rewritten. Parallel execution
enters through an injected mapper whose default is a bounded local process pool — attaching a
cluster-backed one is a later slice. This slice replaces the skeleton stubs for its public
surface and implements top-down through real collaborators to reach a working stage. See the
skeleton issue's Public API contract for exact signatures, and the run-identity seam issue for
the run-id functions it builds on.

### User story

As a **data engineer**, I want to run quality extraction over an explicit list of X-ray files,
so that I can process a new delivery of images without re-reading the images I have already
processed, and know exactly which files a given manifest covers.

### Acceptance criteria

- [ ] Given a listing naming three images, the stage writes one manifest into the destination
      folder containing exactly three records, one per listed image, in listing order.
- [ ] Given a listing containing blank lines, comment lines and surrounding whitespace, only the
      real entries are processed.
- [ ] Given a listing naming images in a folder that also contains images not named in it, the
      unlisted images do not appear in the manifest.
- [ ] Every record in the produced manifest carries an empty split value, its source path, its
      filename, its label taken from the containing folder name, and the extracted statistics.
- [ ] Given a run over the default extractor set, every record carries the Haralick statistics
      requested by configuration, and, when masks are available, the lung-asymmetry statistics.
- [ ] Given a listing whose statistic values place one image outside the configured
      inter-quartile fence, that image's record is flagged excluded with a reason naming the
      offending column, and it is still present in the manifest.
- [ ] Given masks in which one image's lung touches the frame border, that image's record is
      flagged excluded with the out-of-frame reason, and both reasons appear when it is also an
      inter-quartile outlier.
- [ ] Given no masks, no record is flagged out of frame and the stage completes.
- [ ] Given a mask location without an image root to mirror it against, the stage raises
      `ValueError` explaining that both are required together.
- [ ] Running the stage twice over the same listing and the same statistics and filter settings
      writes to the same manifest name; changing a statistics or filter setting writes a
      different one, leaving the first in place.
- [ ] Running the stage twice with different worker counts or batch sizes writes the same
      manifest name and byte-identical manifest content.
- [ ] Two runs over two different listings leave two manifests side by side in the destination
      folder, neither modified by the other.
- [ ] Given a listing in which one image is unreadable and a maximum failure rate of zero, the
      stage raises an extraction failure error naming the failed image and the observed rate,
      and does not write a manifest.
- [ ] Given a listing in which one of four images is unreadable and a maximum failure rate of
      one half, the stage completes, reports one failure and three successes with the
      corresponding failure rate, and the manifest contains the three readable images.
- [ ] Given a listing that resolves to no entries, the stage raises `ValueError` rather than
      writing an empty manifest.
- [ ] Given a listing URI that does not exist, the stage raises `FileNotFoundError` naming it.
- [ ] Given more images than fit in one batch, the stage still produces one record per image and
      never has more work units outstanding at once than its concurrency limit allows.
- [ ] The returned result reports the total, succeeded, failed, failure-rate and excluded counts
      matching the manifest it wrote, and the path of that manifest.
- [ ] Storage options supplied to the stage reach every filesystem access it performs, so a
      filesystem that requires them resolves the listing, the images, the masks and the output.
- [ ] mypy clean; pytest green

### Data flow

```
file listing URI → listed image paths → batches of paths → per-batch records + failures
  → failure-rate check → quality filters (IQR, lung out of frame) → extract manifest
```

### Out of scope

- Assigning splits — records leave this stage with an empty split value on purpose.
- Running batches anywhere other than the local process pool; the injected-mapper seam exists
  from day one but only its local default is wired here.

### Technical notes

- Batching is the unit of work: one mapper unit per batch of image paths, never one per image.
  The local mapper keeps a bounded number of units outstanding rather than submitting every
  batch up front.
- The per-batch function must stay a top-level, picklable function taking only picklable
  arguments, because it crosses a process boundary in the local mapper and a serialization
  boundary in the cluster-backed mappers added later.
- The default worker count comes from the single source of truth in the execution module; no
  other module may re-derive it.
- The quality filters operate on the assembled table of records, after all batches complete —
  the inter-quartile fence is a property of the whole batch of listed images, not of a chunk.
- The configuration hashed into the run id is the output-affecting subset only: mask and image
  roots, the Haralick settings, the filter columns and factor, and the run label. Worker count,
  batch size, failure threshold and runner choice are excluded.

### Design notes

Failure handling flips from silent to loud. Previously an unreadable image was logged at warning
level and dropped, which meant a corrupted delivery could silently shrink a training set. Failed
images are now carried out of the batch function as data, counted, and compared against a
configured maximum failure rate whose default is zero — any failure fails the run unless the
operator explicitly opts into tolerance. The manifest is written only after that check passes, so
a failed run leaves no partially-populated artifact behind for the assign-split stage to pick up.
